test/metabase/test.clj

```diff
@@ -287,6 +290,34 @@
   [clock & body]
   `(do-with-clock ~clock (fn [] ~@body)))
 
+(defn do-with-single-admin-user
+  [attributes thunk]
+  (let [existing-admin-memberships (db/select PermissionsGroupMembership :group_id (:id (perms-group/admin)))
+        _                          (db/simple-delete! PermissionsGroupMembership :group_id (:id (perms-group/admin)))
```

Comment:
you're using `simple-delete!` to avoid delete hooks being called right? 
if so should we add a comment there?

Reply:
now I'm thinking about it, calling `simple-delete` makes this macro incomplete.

if I do 

```clojure
(with-single-admin-user [{id :id}]
  (println (db/select User :is_superuser true)))
```

This does not guarantee to print 1.

I guess it's not important for the immediate usecase, but if you want to make this a generic macro, I think we should fix it.

Reply:
> you're using simple-delete! to avoid delete hooks being called right?

Yes, for the reasons explained below.

> <snip> does not guarantee to print 1.

Good point, fixed.  Rather than futzing with the hooks it seemed easier to just save and restore `is_superuser` directly.  Since this is very behind-the-scenes code I feel okay about doing it directly.